internal/tasks/resourcesync.go

```diff
@@ -97,68 +108,64 @@ func (r *ResourceSync) run(ctx context.Context, log logrus.FieldLogger, rs *mode
-		}
-		listParams.Continue = cont
+		listParams.Continue = listRes.Metadata.Continue
 	}
 
 	fleetsToRemove := fleetsDelta(fleetsPreOwned, fleets)
 
-	r.log.Infof("resourcesync/%s: applying #%d fleets ", rs.Name, len(fleets))
-	err = r.createOrUpdateMultiple(ctx, rs.OrgID, fleets...)
+	r.log.Infof("resourcesync/%s: applying #%d fleets ", *rs.Metadata.Name, len(fleets))
+	err = r.createOrUpdateMultiple(ctx, fleets...)
 	if err == flterrors.ErrUpdatingResourceWithOwnerNotAllowed {
 		err = fmt.Errorf("one or more fleets are managed by a different resource. %w", err)
 	}
 	if len(fleetsToRemove) > 0 {
-		r.log.Infof("resourcesync/%s: found #%d fleets to remove. removing\n", rs.Name, len(fleetsToRemove))
+		r.log.Infof("resourcesync/%s: found #%d fleets to remove. removing\n", *rs.Metadata.Name, len(fleetsToRemove))
 		for _, fleetToRemove := range fleetsToRemove {
-			err := r.store.Fleet().Delete(ctx, rs.OrgID, fleetToRemove, r.callbackManager.FleetUpdatedCallback)
-			if err != nil {
-				log.Errorf("resourcesync/%s: failed to remove old fleet %s/%s. error: %s", rs.Name, rs.OrgID, fleetToRemove, err.Error())
+			status := r.serviceHandler.DeleteFleet(ctx, fleetToRemove)
+			if status.Code != http.StatusOK {
+				log.Errorf("resourcesync/%s: failed to remove old fleet %s. error: %s", *rs.Metadata.Name, fleetToRemove, status.Message)
 				return err
 			}
 		}
 	}
 	rs.AddSyncedCondition(err)
 	if err != nil {
-		log.Errorf("resourcesync/%s: failed to apply resource. error: %s", rs.Name, err.Error())
+		log.Errorf("resourcesync/%s: failed to apply resource. error: %s", *rs.Metadata.Name, err.Error())
 		return err
 	}
-	rs.Status.Data.ObservedGeneration = rs.Generation
-	r.log.Infof("resourcesync/%s: #%d fleets applied successfully\n", rs.Name, len(fleets))
+	rs.Status.ObservedGeneration = rs.Metadata.Generation
+	r.log.Infof("resourcesync/%s: #%d fleets applied successfully\n", *rs.Metadata.Name, len(fleets))
 	return nil
 }
 
-func (r *ResourceSync) createOrUpdateMultiple(ctx context.Context, orgId uuid.UUID, resources ...*api.Fleet) error {
+func (r *ResourceSync) createOrUpdateMultiple(ctx context.Context, resources ...*api.Fleet) error {
 	var errs []error
 	for _, resource := range resources {
-		_, _, err := r.store.Fleet().CreateOrUpdate(ctx, orgId, resource, nil, false, r.callbackManager.FleetUpdatedCallback)
-		if err == flterrors.ErrUpdatingResourceWithOwnerNotAllowed {
-			err = fmt.Errorf("one or more fleets are managed by a different resource. %w", err)
+		_, status := r.serviceHandler.ReplaceFleet(ctx, *resource.Metadata.Name, *resource)
+		if status.Code != http.StatusOK && status.Code != http.StatusCreated {
+			var err error
+			if status.Message == flterrors.ErrUpdatingResourceWithOwnerNotAllowed.Error() {
+				err = fmt.Errorf("one or more fleets are managed by a different resource")
+			} else {
+				err = service.ApiStatusToErr(status)
+			}
+			errs = append(errs, err)
```

Comment:
optional: no need to allocate

```suggestion
			if status.Message == flterrors.ErrUpdatingResourceWithOwnerNotAllowed.Error() {
			    errs = append(errs, fmt.Errorf("one or more fleets are managed by a different resource"))
			} else {
    			    errs = append(errs, service.ApiStatusToErr(status))
			}
```

Reply:
Done